site/content/en/development/design-docs/ipam.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Support node IPAM controller"
+linkTitle: "node IPAM controller"
+type: docs
+weight: 10
+description: >
+Support node IPAM controller in the out-of-tree provider Azure.
+---
+
+## Background
+
+The in-tree node IPAM controller would be moved to the GCE staging repo in the future, 
+[PR](https://github.com/kubernetes/kubernetes/pull/97925). Hence, we need to copy the code to the out-of-tree repo and 
+implement the azure-specific logic here. Specifically, we want to allocate the CIDR for each VMSS and VMAS so every
+instance would get a pod CIDR within the range of the corresponding VMSS or VMAS, instead of the cluster CIDR.
+
+## User experience
+
+There are two kinds of CIDR allocator in the node IPAM controller, which are `RangeAllocator` and `CloudAllocator`.
+The `RangeAllocator` is the default one which allocates the pod CIDR for every node in the range of the cluster CIDR.
+The `CloudAllocator` allocates the pod CIDR for every node in the range of the CIDR on the corresponding VMSS or VMAS.
+The CIDR on the VMSS or VMAS is set by a specific tag `{"VMSet-CIDR": "10.244.1.0/24"}` and every node would get a 
+pod CIDR within the range. Note that the range in the VMSS or VMAS must be within the cluster CIDR, or an error would be 
+thrown. The mask of each pod CIDR could be modified by setting `--node-cidr-mask-size`. For example, 
+if the `--node-cidr-mask-size` is 24 the first node would be allocated with a pod CIDR of "10.244.1.0/24", and the second
+would be allocated "10.244.2.0/24", etc.
+
```

Comment:
Add a comment that `--node-cidr-mask-size` would be the default mask size when the above tag doesn't exist.